src/TriggerBinding/SqlTriggerListener.cs

```diff
@@ -572,10 +577,10 @@ private ScaleStatus GetScaleStatusCore(int workerCount, SqlTriggerMetrics[] metr
             // Check if there is a continuous increase or decrease in count of unprocessed changes.
             bool isIncreasing = true;
             bool isDecreasing = true;
-            for (int index = metrics.Length - minSamplesForScaling; index < metrics.Length - 1; index++)
+            for (int index = 0; index < metrics.Length - 1; index++)
             {
                 isIncreasing = isIncreasing && metrics[index].UnprocessedChangeCount < metrics[index + 1].UnprocessedChangeCount;
-                isDecreasing = isDecreasing && (metrics[index].UnprocessedChangeCount == 0 || metrics[index].UnprocessedChangeCount > metrics[index + 1].UnprocessedChangeCount);
+                isDecreasing = isDecreasing && (metrics[index + 1].UnprocessedChangeCount == 0 || metrics[index].UnprocessedChangeCount > metrics[index + 1].UnprocessedChangeCount);
```

Comment:
This was qualifying samples such as [0, 0, 0, 0, 1] or [0, 4, 3, 2, 1] as strictly decreasing. We only want [1, 0, 0, 0, 0] or [3, 2, 1, 0, 0] to qualify instead.